src/main/java/io/nitric/api/secret/SecretVersion.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package io.nitric.api.secret;
+
+import java.nio.charset.StandardCharsets;
+
+import io.nitric.proto.secret.v1.SecretAccessRequest;
+import io.nitric.proto.secret.v1.SecretAccessResponse;
+import io.nitric.util.Contracts;
+import io.nitric.util.ProtoUtils;
+
+/**
+ * <p>
+ * Provides a Secret Version class. The Secret Version is used to specify a particular secret version stored
+ * with the Secret API. To obtain the latest version of a secret use the <code>Secret.latest()</code> method.
+ * </p>
+ *
+ * @see Secret
+ * @see Secrets
+ */
+public class SecretVersion {
+
+    /** The <code>"latest"</code> secret version number. */
+    public static final String LATEST = "latest";
+
+    final Secret secret;
+    final String version;
+
+    /*
+     * Enforce package builder pattern.
+     */
+    SecretVersion(Secret secret, String version) {
+        Contracts.requireNonNull(secret, "secret");
+        Contracts.requireNonBlank(version, "version");
+
+        this.secret = secret;
+        this.version = version;
+    }
+
+    // Public Methods ---------------------------------------------------------
+
+    /**
+     * Return the version secret.
+     *
+     * @return the version secret
+     */
+    public Secret getSecret() {
+        return secret;
+    }
+
+    /**
+     * Return the secret version.
+     *
+     * @return the secret version
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * <p>
+     * Return the secret version's value data.
+     * </p>
+     *
+     * <pre><code class="code">
+     * String keyVersion = ...
+     *
+     * SecretVersion secretVersion = Secrets.secret("encryption.key").version(keyVersion);
+     *
+     * byte[] keyData = secretVersion.value();
+     * </code></pre>
+     *
+     * @return the secret version's value data
+     */
+    public byte[] value() {
+        var secret = io.nitric.proto.secret.v1.Secret.newBuilder()
+            .setName(getSecret().getName())
+            .build();
+
+        var secretVersion = io.nitric.proto.secret.v1.SecretVersion.newBuilder()
+            .setSecret(secret)
+            .setVersion(getVersion())
+            .build();
+
+        var request = SecretAccessRequest.newBuilder()
+                .setSecretVersion(secretVersion)
+                .build();
+
+        SecretAccessResponse response = null;
+        try {
+            response = Secrets.getServiceStub().access(request);
+        } catch (io.grpc.StatusRuntimeException sre) {
+            throw ProtoUtils.mapGrpcError(sre);
+        }
+
+        var body = response.getValue();
+        var value = (!body.isEmpty()) ? body.toByteArray() : new byte[0];
+
+        return value;
+    }
```

Comment:
I think it would be good to look at potentially standardising this method across across the SDK implementations. I think value is good, but has a very getter/passive feel to it, and doesn't imply to its user that a remote or network based action is taking place, but they're paying per X API calls: https://aws.amazon.com/secrets-manager/pricing/.

In the go SDK I've used `Access()`.

I see you've got convenience methods for retrieving the value as text as well which is great!, maybe we could look at having a single method that returns a secret metadata and contents object (similiar to documents) that has these convenience methods attached to it?

Reply:
Yes standardization is good. In the Java world I would think ```get()``` would be the most natural method name. I find ```Access()``` a bit ambiguous in terms of what is referring to, is this a ACL policy etc.

Reply:
Done